pkg/types/docker_conf.go

```diff
@@ -21,10 +23,26 @@ func GetDockerOption(insecureTlsSkip bool, Platform string) (types.DockerOption,
 	if err := env.Parse(&cfg); err != nil {
 		return types.DockerOption{}, xerrors.Errorf("unable to parse environment variables: %w", err)
 	}
+	credentials := make([]types.Credential, 0)
+	users := strings.Split(cfg.UserName, ",")
+	passwords := strings.Split(cfg.Password, ",")
+	for index, user := range users {
+
+		if len(user) > 0 && index < len(passwords) && len(passwords[index]) > 0 {
+			credentials = append(credentials, types.Credential{
+				UserName: strings.TrimSpace(user),
+				Password: strings.TrimSpace(passwords[index]),
+			})
+		}
+	}
+	if len(credentials) == 0 {
+		credentials = append(credentials, types.Credential{}) // no credential use-case
+	}
```

Comment:
Can we move it here so we don't have to pass `[]types.Credential{{}}` everywhere?
https://github.com/aquasecurity/trivy/blob/4405cd2b01221d5a4e791afdcaa846a23e09e3a0/pkg/fanal/image/image.go#L100-L109